src/dispatch.rs

```diff
@@ -94,8 +94,7 @@ impl ApduDispatch
     fn apdu_type<const S: usize>(apdu: &iso7816::Command<S>) -> RequestType {
         info!("instruction: {:?} {}", apdu.instruction(), apdu.p1);
         if apdu.instruction() == Instruction::Select && (apdu.p1 & 0x04) != 0 {
-            // RequestType::Select(Aid::try_from_slice(apdu.data()).unwrap())
-            RequestType::Select(Aid::new(apdu.data()))
+            Aid::try_new(&apdu.data()).map_or(RequestType::NewCommand, RequestType::Select)
```

Comment:
What do you think about returning `None` instead of `NewCommand`?  I don’t think we should pass a request with an invalid AID to the app.

Reply:
Why not. I think this should be logged in any case.

I'll make it `None` and add a `warn!` log.

Reply:
Ok, I fixed it by adding a `BadCommand` request type that makes the dispatch return an error